internal/flex/flex.go

```diff
@@ -181,6 +197,21 @@ func ApplyToAll[T, U any](s []T, f func(T) U) []U {
 	return v
 }
 
+// ApplyToAllMultiSlice returns a new slice containing the results of applying the function `f` to each element of the original slice `s` and `u`.
+func ApplyToAllMultiSlice[T, U, V any](s []T, u []U, f func(T, U) V) []V {
+	v := make([]V, len(s))
+	if len(s) != len(u) {
+		return v
+	}
+	for i, e := range s {
+		if i <= len(u)-1 {
```

Comment:
do you need a check here?

Reply:
Yes, we don't need that, given I do the check before the loop. Will remove it